File: 4/2/main.go
package main

import (
	"bufio"
	"fmt"
	"net"
	"os"
	"strings"
)

func main() {
	conn, err := net.Dial("tcp", "www.yandex.ru:80")
	if err != nil {
		fmt.Println("Ошибка подключения:", err)
		os.Exit(1)
	}
	defer conn.Close()

	request := "GET / HTTP/1.1\r\nHost: www.yandex.ru\r\nConnection: close\r\n\r\n"
	fmt.Fprintf(conn, request)

	reader := bufio.NewReader(conn)
	for {
		line, err := reader.ReadString('\n')
		if err != nil {
			break
		}
		if strings.HasPrefix(line, "set-cookie:") || strings.HasPrefix(line, "Set-Cookie:") {
			fmt.Println("Cookie:", line)
			continue
		}
		fmt.Print(line)
	}
}

File: 4/python/2/main.py
import socket

def main():
    host = "www.yandex.ru"
    port = 80

    try:
        conn = socket.create_connection((host, port))
    except Exception as e:
        print("Ошибка подключения:", e)
        exit(1)
    request = (
        "GET / HTTP/1.1\r\n"
        "Host: www.yandex.ru\r\n"
        "Connection: close\r\n"
        "\r\n"
    )
    conn.sendall(request.encode('utf-8'))

    with conn.makefile("r", encoding="utf-8") as response:
        for line in response:
            if line.lower().startswith("set-cookie:"):
                print("Cookie:", line.strip())
                continue
            print(line, end="")

    conn.close()

if __name__ == "__main__":
    main()
